### PR TITLE
Add Dockerized Dev Environment

### PR DESCRIPTION
**Pull request by schreiaj**
# Pull Request

## Description
Enables dev environment to be spun up with `docker compose -f dev-env.yaml up` enables automatic rebuilding on changes and doesn't require environment setup

## Related Issues
General Onboarding

## Additional Notes
Previous approaches work as normal, this is just to simplify setup on new machines. 

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [x] I have minimized the number of reviewers to include only those essential for the review.
- [x] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: OCR/frontend/dev-dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+COPY . /app
+
+WORKDIR /app
+RUN npm install 
+
+
+# CMD npm run dev -- --host
+CMD vite
```
Comment (zdeveloper) [inline OCR/frontend/dev-dockerfile:13]: what does the `vite` cmd do? is it different from the command in the docker compose yaml

Comment (schreiaj) [reply OCR/frontend/dev-dockerfile]: Oops, great catch - in the compose I overwrite the command so I never really tested it without compose running it. Just updated it to be correct so you can run the just this one instance if you want instead without the compose (not that I'd suggest it)

**Review by zdeveloper — APPROVED** [review]
